File: svelte-typescript-app/selenium/todo_page.py
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.by import By

from locators import *

class BasePage(object):
    def __init__(self,driver):
        self.driver = driver
#get the title of the page
class todo(BasePage):
    def is_title_matching(self):
        return 'TODO' in self.driver.title

    #create new todo , by entering the new id , its task and click on create button
    def createNewTodo(self):
         wait = WebDriverWait(self.driver, 60)
         wait.until(EC.visibility_of_element_located((By.XPATH, todoPage.IdTextField))).send_keys('10')

         self.driver.find_element(By.XPATH, todoPage.TaskTextField).send_keys('ten')
         self.driver.find_element(By.XPATH, todoPage.CreateButton).click()
    
     #delete todo , by entering the id and click on delete button
    def deleteExistingTodo(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.DeleteMark).click()
    
    #update  todo , by entering the id and its task and click on update button
    def upadateTodo(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.IdTextField).send_keys('1')
         self.driver.find_element(By.XPATH, todoPage.TaskTextField).send_keys('DoaaSaber')
         self.driver.find_element(By.XPATH, todoPage.UpdateButton).click()
    
     #get todo , by entering the id and click on get button
    def getbyID(self):
        wait = WebDriverWait(self.driver, 60)
        self.driver.find_element(By.XPATH, todoPage.IdTextField).send_keys('1')
        self.driver.find_element(By.XPATH, todoPage.GetButton).click()
    
     #get todo , by entering un-existing id and click on get button
    def getUnExisting(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.IdTextField).send_keys('16')
         self.driver.find_element(By.XPATH, todoPage.GetButton).click()
    
    #create new todo , by entering the new id and click on create button
    def createNewTodoWithouttask(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.IdTextField).send_keys('56')
         self.driver.find_element(By.XPATH, todoPage.CreateButton).click()
    
     #delete todo , by entering un-existing id and click on delete button
    def deleteUnExisting(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.IdTextField).send_keys('98')
         self.driver.find_element(By.XPATH, todoPage.DeleteButton).click()
    
    #update  todo , by entering un-existing id and its task and click on update button
    def upadateUnExistingTodo(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.IdTextField).send_keys('115')
         self.driver.find_element(By.XPATH, todoPage.TaskTextField).send_keys('DoaaSaber')
         self.driver.find_element(By.XPATH, todoPage.UpdateButton).click()
      
     #create new todo , by entering the new id(begain with character) , its task and click on create button
    def createwithChar(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.IdTextField).send_keys('hello')
         self.driver.find_element(By.XPATH, todoPage.TaskTextField).send_keys('world')
         self.driver.find_element(By.XPATH, todoPage.CreateButton).click()
    
     #create new todo , by entering the existing id ,its task and click on create button
    def createWithExistingID(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.IdTextField).send_keys('1')
         self.driver.find_element(By.XPATH, todoPage.TaskTextField).send_keys('Amira')
         self.driver.find_element(By.XPATH, todoPage.CreateButton).click()
    
    #click on the checkbox to toggle the "done" 
    def convertCheckBox(self):
         wait = WebDriverWait(self.driver, 60)
         self.driver.find_element(By.XPATH, todoPage.CheckMark).click()
    

File: svelte-typescript-app/cypress/pages-objects/todo_page.js
class todo{
    getTitle(){
        return cy.get('h2.svelte-1odsl5c')
    }
     getIdInput(){
        return cy.get('#inputID')
    }
     getTaskInput(){
        return cy.get('#inputTask')
    }
     getSubmitButton(){
        return  cy.get('[type="create"]')
    }
     getUpdateButton(){
        return cy.get('[type="update"]')
    }
     getDeleteButton(){
        return cy.get('[type="delete"]')
    }
     getButton(){
        return cy.get('[type="get"]')
        
    }
     getDeleteMark(){
       return cy.get(':nth-child(5) > .btn > i')
    }

    getcheckMark(){
        return cy.get(':nth-child(6) > input')
    }



    //create new todo , by entering the new id , its task and click on create button
    createNewtodo(){

        this.getTitle().contains("TODO APP")
        this.getIdInput().type("166")
        this.getTaskInput().type("eaten")
        this.getSubmitButton().click()
    }
    //delete todo , by entering the id and click on delete button

    deleteExistingTodo(){
        this.getDeleteMark().click()
    }

    //update  todo , by entering the id and its task and click on update button

    updateExisitingTodo(){
         this.getIdInput().type("1")
         this.getTaskInput().type("DoaaSaber")
         this.getUpdateButton().click()
        }
        
        //get todo , by entering the id and click on get button

    getExisitingTodo(){
        this.getIdInput().type("1")
        this.getButton().click()}

   //get todo , by entering un-existing id and click on get button
 
    getUnExisitingTodo(){
    this.getIdInput().clear({force: true})
    this.getIdInput().type("15")
    this.getButton().click()

    }


    //create new todo , by entering the new id and click on create button

    createNewTodoWithoutEnteredTask(){
    this.getIdInput().type("9")
    this.getSubmitButton().click()}

    //delete todo , by entering un-existing id and click on delete button

    deleteUnExistingTodo(){
    this.getIdInput().type("36")
    this.getDeleteButton().click()
    }

    //update  todo , by entering un-existing id and its task and click on update button

    updateUnExistingTodo(){
        this.getIdInput().type("88")
        this.getTaskInput().type("DoaaSaber")
        this.getUpdateButton().click()
    }

    //create new todo , by entering the new id(begain with character) , its task and click on create button

    createNewTodoWithCharId(){
        this.getIdInput().type("BB")
        this.getTaskInput().type("jj")
        this.getSubmitButton().click()
    }

    //create new todo , by entering the existing id ,its task and click on create button

    createNewTodoWithExisitingId(){
        this.getIdInput().type("1")
        this.getTaskInput().type("amira")
        this.getSubmitButton().click()
    }

    //click on the checkbox to toggle the "done" 

    checkboxMark(){
        this.getcheckMark().click()
        this.getIdInput().type(" ")
        this.getIdInput().type(1)
        this.getButton()
    }
    
}
export default new todo();